src/systems/filecoin_markets/deal/_index.md

```diff
@@ -65,3 +65,34 @@ On every PoSt Submission, the following steps happen.
 # Deal Flow
 
 {{< diagram src="diagrams/deal-flow.mmd.svg" title="Deal Flow Sequence Diagram" >}}
+
+# Deal States
+
+All on-chain economic activities in Filecoin start with the deal. This section aims to explain different states of a deal and their relationship with other concepts in the protocol such as Power, Payment, and Collaterals.
+
+A deal has the following states:
+
+- `Unpublished`: the deal has yet to be posted on chain.
+- `Published`: the deal has been published and accepted by the chain but is not yet active as the sector containing the deal has not been proven.
+- `Active`: the deal has been proven and not yet expired.
+- `Deleted`: the deal or the sector containing the deal has expired or the sector containing the deal has been terminated because of faults.
+
+Note that `Unpublished` and `Deleted` states not tracked on chain. To reduce on-chain footprint, an `OnchainDeal` struct is created when a deal is published and it keeps track of a `LastPaymentEpoch` which defaults to zero when a deal is in the `Published` state. A deal transitions into the `Active` state when `LastPaymentEpoch` is positive.
+
+The following describes how a deal transitions between its different states.
+
+- `Unpublished -> Published`: this is triggered by `StorageMarketActor.PublishStorageDeals` which validates new storage deals, lock necessary funds, generate deal IDs, and register the storage deals in `StorageMarketActor`.
+- `Published -> Deleted`: this is triggered by `StorageMinerActor.ProveCommitSector` during InteractivePoRep when the elapsed Epoch between PreCommit and ProveCommit messages exceeds `MAX_PROVE_COMMIT_SECTOR_EPOCH`. ProveCommitSector will also trigger garbage collection on the list of Published storage deals.
+- `Published -> Active`: this is triggered by `ActivatateStorageDeals` after successful `StorageMinerActor.ProveCommitSector`. It is okay for the StorageDeal to have already started at this point but it must not have expired.
+- `Active -> Deleted`: this can happen in the following conditions:
+  - The deal itself has expired. This is triggered by `StorageMinerActorCode._submitPowerReport` which is called whenever a PoSt is submitted. Power associated with the deal will be lost, collaterals returned, and all remaining storage fees unlocked.
+  - The sector containing the deal has expired. This is triggered by `StorageMinerActorCode._submitPowerReport` which is called whenver a PoSt is submitted. Power associated with the deals in the sector will be lost, collaterals returned, and all remaining storage fees unlocked.
```

Comment:
unlocked or paid out?

Reply:
again can the sector expire without all deals expiring? if not delete this bullet.

Reply:
unlocked. Miners need to submit WithdrawBalance to pay out. To the second point, sector expires when all its deals expire, we can delete this bullet but wanted to make it clearer.